packages/airnode-protocol-v1/contracts/monetization/RequesterAuthorizerWhitelisterWithTokenDeposit.sol

```diff
@@ -158,6 +189,73 @@ contract RequesterAuthorizerWhitelisterWithTokenDeposit is
                     false
                 );
         }
+    }
+
+    /// @notice Withdraws tokens previously deposited for the requester to be
+    /// whitelisted for the Airnode–endpoint pair on the chain
+    /// @param airnode Airnode address
+    /// @param chainId Chain ID
+    /// @param endpointId Endpoint ID
+    /// @param requester Requester address
+    function withdrawTokens(
+        address airnode,
+        uint256 chainId,
+        bytes32 endpointId,
+        address requester
+    ) external override onlyNonBlockedRequester(airnode, requester) {
+        TokenDeposits
+            storage tokenDeposits = airnodeToChainIdToEndpointIdToRequesterToTokenDeposits[
+                airnode
+            ][chainId][endpointId][requester];
+        uint256 tokenWithdrawAmount = tokenDeposits.depositorToAmount[
+            msg.sender
+        ];
+        require(tokenWithdrawAmount != 0, "Sender has not deposited tokens");
+        tokenDeposits.depositorToAmount[msg.sender] = 0;
+        uint256 earliestWithdrawalTime = tokenDeposits
+            .depositorToEarliestWithdrawalTime[msg.sender];
+        if (withdrawalLeadTime != 0) {
```

Comment:
only ` if (withdrawalLeadTime != 0)` is probably sufficient here along with the `require()` I think we don't need the if below. The else could lead this if

Reply:
I don't understand this. Would that implement the same logic?

Reply:
in a way, take this for example:
```
if (withdrawalLeadTime != 0) {
    require(
            earliestWithdrawalTime <= block.timestamp,
            "Intent not signaled or not withdrawal time yet"
        );
        uint256 tokenDepositsCount = tokenDeposits.count;
        emit WithdrewTokens(
            airnode,
            chainId,
            endpointId,
            requester,
            msg.sender,
            tokenDepositsCount,
            tokenWithdrawAmount
        );
} else 
```
Essentially only check `earliestWithdrawalTime` if there is a lead time otherwise don't

Reply:
There, I'm also checking `earliestWithdrawalTime` to see if I need to decrement the deposit count or not. So if there was a `withdrawalLeadTime` and someone signaled withdrawal intent but then `withdrawalLeadTime` was set to zero and the user wants to withdraw, we shouldn't decrement the deposit count twice.

Reply:
I think I addressed this in https://github.com/api3dao/airnode/pull/926/commits/cde8b1b2d23fc547cce2197054a5bb1c03a34caa